build_tools/android/upload_benchmarks_to_dashboard.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python3
+# Lint as: python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Upload benchmark results to IREE Benchmark Dashboards.
+
+This script is meant to be used by Buildkite for automation.
+
+Example usage:
+  # Export necessary environment variables:
+  export IREE_DASHBOARD_URL=...
+  export IREE_DASHBOARD_API_TOKEN=...
+  # Then run the script:
+  python3 upload_benchmarks.py /path/to/benchmark/json/file
+"""
+
+import argparse
+import json
+import os
+import re
+import requests
+import subprocess
+import time
+
+from typing import Any, Dict, Optional
+
+from common.benchmark_description import (BenchmarkInfo, BenchmarkResults,
+                                          get_output)
+
+IREE_GITHUB_COMMIT_URL_PREFIX = 'https://github.com/google/iree/commit'
+IREE_PROJECT_ID = 'IREE'
+
+# A non-exhaustive list of models and their source URLs.
+# For models listed here we can provide a nicer description for them on
+# webpage.
+IREE_TF_MODEL_SOURCE_URL = {
+    'MobileNetV2':
+        'https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV2',
+    'MobileNetV3Small':
+        'https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV3Small',
+    'MobileBertSquad':
+        'https://github.com/google-research/google-research/tree/master/mobilebert',
+}
+
+
+def get_model_description(benchmark_info: BenchmarkInfo) -> Optional[str]:
+  """Gets the model description for the given benchmark."""
+  url = None
+  name = benchmark_info.model_name
+  if benchmark_info.model_source == "TensorFlow":
+    url = IREE_TF_MODEL_SOURCE_URL.get(name)
+  if url is not None:
+    description = f'{name} from <a href="{url}">{url}</a>.'
+    return description
+  return None
+
+
+def get_git_commit_hash(commit: str, verbose: bool = False) -> str:
+  """Gets the commit hash for the given commit."""
+  return get_output(['git', 'rev-parse', commit],
+                    cwd=os.path.dirname(os.path.realpath(__file__)),
```

Comment:
Can you bust this out as a module level constant `THIS_DIRECTORY` (or equiv)? Here and below.

Reply:
Done.